scheduling/storage_providers/memory.go

```diff
@@ -37,13 +70,109 @@ func (m *MemoryStorageProvider) Get(key string, prefix bool) ([]string, error) {
 	}
 
 	var result []string
-	for k, v := range m.storage {
+	var keys []string
+	for k, _ := range m.storage {
 		if strings.HasPrefix(k, key) {
-			result = append(result, v)
+			keys = append(keys, k)
+		}
+	}
+
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		if strings.HasPrefix(k, key) {
+			result = append(result, m.storage[k])
 		}
 	}
 	if len(result) == 0 {
 		return nil, &KeyNotFoundError{Key: key}
 	}
 	return result, nil
 }
+
+func (m *MemoryStorageProvider) ListKeys(prefix string) ([]string, error) {
+	var result []string
+	for k, _ := range m.storage {
+		if strings.HasPrefix(k, prefix) {
+			result = append(result, k)
+		}
+	}
+	sort.Strings(result)
+
+	return result, nil
+}
+
+type LockObject struct {
+	ID      string
+	Channel chan error
+}
+
+func (m *MemoryStorageProvider) Lock(id string, key string, lockChannel chan error) (LockObject, error) {
```

Comment:
can we create the lock internally and return it? Dont want to have to pass it in